ldclient/impl/datasourcev2/streaming.py

```diff
@@ -0,0 +1,374 @@
+                continue
+
+            try:
+                update = self._process_message(action, change_set_builder)
+                if update is not None:
+                    yield update
+            except json.decoder.JSONDecodeError as e:
+                log.info(
+                    "Error while handling stream event; will restart stream: %s", e
+                )
+                # TODO(sdk-1409)
+                # self._sse.interrupt()
+
+                (should_continue, update) = self._handle_error(e)
+                if update is not None:
+                    yield update
+                if not should_continue:
+                    break
+            except Exception as e:  # pylint: disable=broad-except
+                log.info(
+                    "Error while handling stream event; will restart stream: %s", e
+                )
+                # TODO(sdk-1409)
+                # self._sse.interrupt()
+
+                yield Update(
+                    state=DataSourceState.INTERRUPTED,
+                    error=DataSourceErrorInfo(
+                        DataSourceErrorKind.UNKNOWN, 0, time(), str(e)
+                    ),
+                    revert_to_fdv1=False,
+                    # to signal that the client should revert to fdv1.
+                    environment_id=None,  # TODO(sdk-1410)
+                )
+
+            # TODO(sdk-1408)
+            # if update is not None:
+            #     self._record_stream_init(False)
+
+            # if self._data_source_update_sink is not None:
+            #     self._data_source_update_sink.update_status(
+            #         DataSourceState.VALID, None
+            #     )
+
+            # if not self._ready.is_set():
+            #     log.info("StreamingUpdateProcessor initialized ok.")
+            #     self._ready.set()
+
+        # TODO(sdk-1409)
+        # self._sse.close()
+
+    # TODO(sdk-1409)
+    # def stop(self):
+    #     self.__stop_with_error_info(None)
+    #
+    # def __stop_with_error_info(self, error: Optional[DataSourceErrorInfo]):
+    #     log.info("Stopping StreamingUpdateProcessor")
+    #     self._running = False
+    #     if self._sse:
+    #         self._sse.close()
+    #
+    #     if self._data_source_update_sink is None:
+    #         return
+    #
+    #     self._data_source_update_sink.update_status(DataSourceState.OFF, error)
+
+    # pylint: disable=too-many-return-statements
+    def _process_message(
+        self, msg: Event, change_set_builder: ChangeSetBuilder
+    ) -> Optional[Update]:
+        """
+        Processes a single message from the SSE stream and returns an Update
+        object if applicable.
+
+        This method may raise exceptions if the message is malformed or if an
+        error occurs while processing the message. The caller should handle these
+        exceptions appropriately.
+        """
+        if msg.event == EventName.HEARTBEAT:
+            # https://tinyurl.com/3ne69k2t
```

Comment:
Not sure about this. That said I would avoid a shortener after the `goo.gl` situation.